crates/metrics/src/config.rs

```diff
@@ -0,0 +1,99 @@
+use once_cell::sync::Lazy;
+use strum::IntoEnumIterator;
+use strum_macros::{
+    Display,
+    EnumIter,
+    EnumString,
+};
+
+#[derive(Debug, Display, Clone, Copy, PartialEq, EnumString, EnumIter)]
+#[strum(serialize_all = "lowercase")]
+pub enum Module {
+    Importer,
+    P2P,
+    Producer,
+    TxPool, /* TODO[RC]: Not used. Add support in https://github.com/FuelLabs/fuel-core/pull/2321 */
+    GraphQL, // TODO[RC]: Not used... yet.
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct Config(Vec<Module>);
+impl Config {
+    pub fn is_enabled(&self, module: Module) -> bool {
+        !self.0.contains(&module)
+    }
+}
+
+impl std::convert::From<&str> for Config {
+    fn from(s: &str) -> Self {
+        if s == "all" {
+            return Self(Module::iter().collect())
+        }
+        Self(
+            s.split(',')
+                .filter_map(|s| s.parse::<Module>().ok())
+                .collect(),
+        )
+    }
+}
```

Comment:
I think we want to have `TryFrom` and return an error=)

Reply:
Unfortunately, `clap` doesn't like `TryFrom`.


But it can work with `FromStr`, but there is other limitation that I couldn't overcome, ie. I had to use `String` [here](https://github.com/FuelLabs/fuel-core/pull/2310/commits/a128622777e7237bcf7bffa16e0a10055c2e7321#diff-a347decffd707ec63ac973b354ab47ae78ea78b7b81e460b8d4f442f6dc9a519R37), instead of [dedicated error type](https://github.com/FuelLabs/fuel-core/pull/2310/commits/a128622777e7237bcf7bffa16e0a10055c2e7321#diff-a347decffd707ec63ac973b354ab47ae78ea78b7b81e460b8d4f442f6dc9a519R10).

Updated in https://github.com/FuelLabs/fuel-core/pull/2310/commits/a128622777e7237bcf7bffa16e0a10055c2e7321.

Reply:
Making this fallible caused some test to panic. Investigating...

Reply:
Fixed with https://github.com/FuelLabs/fuel-core/pull/2310/commits/bee9c6fe93df5a30e48c807eea5c25cb5801f984. Not a perfect fix, but I can't make `clap` happy with other solutions.